fv3/io/module_wrt_grid_comp.F90

```diff
@@ -4761,6 +4510,244 @@ subroutine compute_fields_checksum(bundle, rc)
+                                     typekind=ESMF_TYPEKIND_R4, &
+                                     staggerloc=ESMF_STAGGERLOC_CENTER, &
+                                     rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        call ESMF_FieldGet(dst_field, localDeCount=localDeCount, rc=rc);
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        if (localDeCount > 0) then
+          call ESMF_FieldGet(dst_field, farrayPtr=ptr, rc=rc);
+          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          ptr = 0.0
+        end if
+
+        dst_status_field = ESMF_FieldCreate(dst_grid, ESMF_TYPEKIND_I4, name='dst_status_field', rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        srcTermProcessing = 0
+
+        call ESMF_FieldRegridStore(srcField=src_field, &
+                                   dstField=dst_field, &
+                                   regridmethod=ESMF_REGRIDMETHOD_BILINEAR, &
+                                   routehandle=routehandle_mask, &
+                                   unmappedaction=ESMF_UNMAPPEDACTION_IGNORE, &
+                                   srcTermProcessing=srcTermProcessing, &
+                                   dstStatusField=dst_status_field, &
+                                   rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        call ESMF_FieldRegrid(src_field, dst_field, &
+                              routehandle=routehandle_mask, &
+                              termorderflag=ESMF_TERMORDER_SRCSEQ, &
+                              zeroregion=ESMF_REGION_SELECT, rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        call ESMF_FieldGet(dst_field, localDeCount=localDeCount, rc=rc);
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        if (localDeCount > 0) then
+          call ESMF_FieldGet(dst_field, farrayPtr=ptr, rc=rc);
+          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+          call ESMF_FieldGet(dst_status_field, farrayPtr=ptr_dst_status, rc=rc);
+          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+          istart = lbound(ptr,1)
+          iend   = ubound(ptr,1)
+          jstart = lbound(ptr,2)
+          jend   = ubound(ptr,2)
+          do jg=jstart, jend
+            do ig=istart, iend
+              if (ptr(ig,jg) == 0.0 .and. ptr_dst_status(ig,jg) /= ESMF_REGRIDSTATUS_OUTSIDE) then
+                write(0,*)'remap warning: ', ig,jg,ptr(ig,jg), ptr_dst_status(ig,jg)
```

Comment:
I am not sure if we need to keep this section, if this happens, then something is wrong with ESMF.
Maybe more details: "Remap warning - not mapped but not outside source grid: "
line 4698: "Remap warning - mapped but ESMF status does not show mapped: "

Reply:
I prefer removing it.